react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,14 +109,48 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => void;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _onFinalizeServerURL() {
+        const { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(\\w+(?:s)?://)?(.+)$');
+            const urlComponents = urlRegExp.exec(serverURL.toLowerCase());
+
+            if (!urlComponents[1] || !urlComponents[1].startsWith('http')) {
+                try {
+                    this._onChangeServerURL(
+                        new URL(`https://${urlComponents[2]}`).toString()
```

Comment:
Iff the given URL contained HTTP (not S) we should preserve that. beta.meet.jit.si, works over HTTP for example.

Reply:
Not sure what you mean, http:// protocol is preserved

Reply:
Oops, I miss read, disregard this.